File: 997/997.js
// Adapted from https://raw.githubusercontent.com/cirosantilli/project-euler-solutions/refs/heads/master/solvers/997.py

function compute(px, py, pz) {
    return 3n * (1n << (px + py + pz - 1n)) * ((1n << px) + (1n << py) + (1n << pz) - 4n);
}

function main() {
    console.assert(compute(1n, 1n, 1n) === 24n);
    console.assert(compute(2n, 3n, 4n) === 18432n);
    console.log(compute(9n, 10n, 11n).toString());
}

main();

File: 997/997.cs
// Adapted from https://raw.githubusercontent.com/cirosantilli/project-euler-solutions/refs/heads/master/solvers/997.py

using System;

class Program
{
    static long Compute(int px, int py, int pz)
    {
        return 3L * (1L << (px + py + pz - 1)) * ((1L << px) + (1L << py) + (1L << pz) - 4L);
    }

    static void Main()
    {
        System.Diagnostics.Debug.Assert(Compute(1, 1, 1) == 24L);
        System.Diagnostics.Debug.Assert(Compute(2, 3, 4) == 18432L);
        Console.WriteLine(Compute(9, 10, 11));
    }
}